### PR TITLE
Removing yca from CPP client

### PR DESCRIPTION
**Pull request by jai1**
### Motivation

Want to reduce yca usage in pulsar

### Modifications

Removed code under PULSAR_ENABLE_DEPRECATED_METHOD

### CODE REVIEW

**Comment by msb-at-yahoo** [comment]
what's the plan for ABI compat break? force a rebuild to take the new client?

**Comment by msb-at-yahoo** [comment]
👍

**Comment by jai1** [comment]
@merlimat - Changes for #241 are incomplete since we don't have the macro in Auth.cc so if you try to compile the code with -DPULSAR_ENABLE_DEPRECATED_METHOD it will fail since there is no implementation for 
- AuthData 
- static AuthDataPtr Disabled();
- static AuthDataPtr YcaV1(const std::string& ycaAppId);

@msb-at-yahoo - we are planning to maintain an internal branch with changes for "Auth.h" and "Auth.cc", we will pull changed from OS master and push to the internal branch whenever we build a tag.

Since these Auth.* files are rarely changed we will rarely have to do a manual merge.

**Comment by merlimat** [comment]
@jai1 Fine with me.. I was also skeptic of feasibility of source compatibility. Please make sure to check in with @saandrews .

**Comment by msb-at-yahoo** [comment]
makes sense.  longer term we should look at something influenced by SASL so that support for internal auth protocols don't break the ABI.

On Monday, March 13, 2017, 2:12:40 PM PDT, jai1 <notifications@github.com> wrote:
@merlimat - Changes for #241 are incomplete since we don't have the macro in Auth.cc so if you try to compile the code with -DPULSAR_ENABLE_DEPRECATED_METHOD it will fail since there is no implementation for
   
   - AuthData
   - static AuthDataPtr Disabled();
   - static AuthDataPtr YcaV1(const std::string& ycaAppId);

@msb-at-yahoo - we are planning to maintain an internal branch with changes for "Auth.h" and "Auth.cc", we will pull changed from OS master and push to the internal branch whenever we build a tag.

Since these Auth.* files are rarely changed we will rarely have to do a manual merge.

**Comment by merlimat** [comment]
> makes sense.  longer term we should look at something influenced by SASL so that support for internal auth protocols don't break the ABI.

It's already possible, the new auth api takes a "plugin" library as a parameter, so you can pass in a new auth implementation.

The problem was related to the earlier C++ auth api that was having YCA as an always viable option.

**Comment by jai1** [comment]
@merlimat @msb-at-yahoo - I think we already broke the ABI by changing 
`static AuthDataPtr Disabled(); 
`TO
`static AuthenticationPtr Disabled();
`
Since AuthData is not of the same size as Authentication class

Trying to fix it without having to change Client.cc

**Comment by merlimat** [comment]
> I think we already broke the ABI by changing

ABI was broker on renaming the c++ namespace from `cms` to `pulsar`

**Comment by jai1** [comment]
Please don't merge the branch I will me making more changes to fix the broken ABI so that we can compile by adding -Dpulsar=cms

File: pulsar-client-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -57,9 +57,16 @@ set(COMMON_LIBS
 
 link_directories(${CMAKE_BINARY_DIR}/lib)
 
+set(LIB_NAME $ENV{PULSAR_LIBRARY_NAME})
+if (NOT LIB_NAME)
+    set(LIB_NAME pulsar)
+endif(NOT LIB_NAME)
+
+message("Using library name: " ${LIB_NAME})
```
Comment (saandrews) [inline pulsar-client-cpp/CMakeLists.txt]: Why do we need this message?

File: pulsar-client-cpp/lib/CMakeLists.txt [hunk 2]
```diff
@@ -16,15 +16,18 @@
...
 file(GLOB PULSAR_SOURCES *.cc lz4/*.c checksum/*.cc)
 
-execute_process(COMMAND cat ../pom.xml COMMAND xmllint --format - COMMAND sed "s/xmlns=\".*\"//g" COMMAND xmllint --stream --pattern /project/version --debug - COMMAND grep -A 2 "matches pattern" COMMAND grep text COMMAND sed "s/.* [0-9] //g" OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE PV)
+set(PV $ENV{PULSAR_LIBRARY_VERSION})
+if (NOT PV)
+    execute_process(COMMAND cat ../pom.xml COMMAND xmllint --format - COMMAND sed "s/xmlns=\".*\"//g" COMMAND xmllint --stream --pattern /project/version --debug - COMMAND grep -A 2 "matches pattern" COMMAND grep text COMMAND sed "s/.* [0-9] //g" OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE PV)
+endif(NOT PV)
 
-set (CMAKE_CXX_FLAGS " ${CMAKE_CXX_FLAGS} -msse4.2 -mpclmul -D_PULSAR_VERSION_=\\\"${PV}\\\"")
+message("Version: " ${PV})
```
Comment (saandrews) [inline pulsar-client-cpp/lib/CMakeLists.txt]: Remove message?

File: pulsar-client-cpp/include/pulsar/Message.h [hunk 3]
```diff
@@ -26,16 +26,11 @@
...
 
 namespace pulsar {
-namespace proto {
-class CommandMessage;
-class MessageMetadata;
-class SingleMessageMetadata;
-}
-}
-
-using namespace pulsar;
```
Comment (jai1) [inline pulsar-client-cpp/include/pulsar/Message.h]: Removed using namespace pulsar;

**Pull request merged**